File: app/src/main/java/vgame/ir/view/component/AwsomeTextCell.java
package vgame.ir.view.component;
import android.content.Context;
import android.graphics.Typeface;
import androidx.appcompat.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.Gravity;

import vgame.ir.utils.Common;


public class AwsomeTextCell extends AppCompatTextView {

    public AwsomeTextCell(Context context) {
        super(context);
        setCustomFont();
    }

    public AwsomeTextCell(Context context, AttributeSet attrs) {
        super(context, attrs);
        setCustomFont();
        setGravity(Gravity.CENTER);
    }

    public void setCustomFont() {
        int style = getTypeface().getStyle();
        if(style == Typeface.BOLD){
            Typeface myTypeface = Typeface.createFromAsset(getContext().getAssets(), Common.FontsAddress.AWSOME_BOLD);
            setTypeface(myTypeface);
        }else {
            Typeface myTypeface = Typeface.createFromAsset(getContext().getAssets(), Common.FontsAddress.AWSOME);
            setTypeface(myTypeface);
        }

    }
}

File: app/src/main/java/vgame/ir/view/component/AwsomeTextCell.kt
package vgame.ir.view.component

import android.content.Context
import android.graphics.Typeface
import androidx.appcompat.widget.AppCompatTextView
import android.util.AttributeSet
import android.view.Gravity

import vgame.ir.utils.Common


class AwsomeTextCell : AppCompatTextView {

    constructor(context: Context) : super(context) {
        setCustomFont()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        setCustomFont()
        gravity = Gravity.CENTER
    }

    fun setCustomFont() {
        val style = typeface.style
        if (style == Typeface.BOLD) {
            val myTypeface = Typeface.createFromAsset(context.assets, Common.FontsAddress.AWSOME_BOLD)
            typeface = myTypeface
        } else {
            val myTypeface = Typeface.createFromAsset(context.assets, Common.FontsAddress.AWSOME)
            typeface = myTypeface
        }

    }
}

